zsh-aws-vault.plugin.zsh

```diff
@@ -56,47 +113,49 @@ function avli() {
   local browser="$(_find_browser)"
 
   if _using_osx ; then
+    local browser_profile_path=$(_get_browser_profile_path $browser $1)
     case $browser in
+      # TODO: this doesn't seem to work on MacOS Firefox - I think the profile needs to be in the "expected" location, as we were doing before...
+      # I need to dig more into this.
       org.mozilla.firefox)
-        # Ensure a profile is created (can run idempotently) and launch it as a disowned process
-        /Applications/Firefox.app/Contents/MacOS/firefox --CreateProfile $1 2>/dev/null && \
-        /Applications/Firefox.app/Contents/MacOS/firefox $AWS_VAULT_PL_BROWSER_LAUNCH_OPTS --no-remote -P $1 "${login_url}" 2>/dev/null &!
+        /Applications/Firefox.app/Contents/MacOS/firefox $AWS_VAULT_PL_BROWSER_LAUNCH_OPTS --no-remote -P $browser_profile_path $login_url 2>/dev/null && \
+        _maybe_clean_up_browser_profile "${browser_profile_path}" &!
         ;;
       org.mozilla.firefoxdeveloperedition)
-        /Applications/Firefox\ Developer\ Edition.app/Contents/MacOS/firefox --CreateProfile $1 2>/dev/null && \
-        /Applications/Firefox\ Developer\ Edition.app/Contents/MacOS/firefox $AWS_VAULT_PL_BROWSER_LAUNCH_OPTS --no-remote -P $1 "${login_url}" 2>/dev/null &!
+        /Applications/Firefox\ Developer\ Edition.app/Contents/MacOS/firefox $AWS_VAULT_PL_BROWSER_LAUNCH_OPTS --no-remote -P $browser_profile_path $login_url 2>/dev/null && \
+          _maybe_clean_up_browser_profile "${browser_profile_path}" &!
         ;;
```

Comment:
This is listed as a TODO on the PR description. I need to spend some time seeing if we can pass an arbitrary profile path. My initial testing did not work. I'd like to sync up this behavior with the other browsers, and this appears to work on Linux, so hopefully I can figure it out.